src/averageAges.js

```diff
@@ -55,7 +83,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let sum = 0;
+  let sumChildren = 0;
+  let mother = [];
+
+  const result = people.filter(person =>
+    (people.filter(children => person.name === children.mother)).length > 0);
+
+  people.forEach(element => {
```

Comment:
<img width="250" alt="image" src="https://user-images.githubusercontent.com/66749919/189191472-6d18669b-559b-4269-b9c2-eacf44633d43.png">

Reply:
not fixed